### ISSUE TITLE
Storage.UploadFromStream results in error

### ISSUE THREAD

**Issue by gorepj** [SchneiderInfosystems/FB4D#3]
**Describe the bug**
The Intro demo program uses Storage.UploadSynchronousFromStream to demonstrate upload of a file.
This works perfectly. I need to implement the asynchronous alternative for Android use.

When I replace the Call using Storage.UploadFromStream I get an error every time 

"Invalid HTTP method/URL pair."

**To Reproduce**
1. Open the Intro demo project
2. From line 704 rplace
    try
      fs := TFileStream.Create(OpenDialog.FileName, fmOpenRead);
      try
        fFirestoreObject := Storage.UploadSynchronousFromStream(fs, ObjectName,ContentType);
        ShowFirestoreObject;
      finally
        fs.Free;
      end;
    finally
      Storage.Free;
    end;
  end;

with
    try
      fs := TFileStream.Create(OpenDialog.FileName, fmOpenRead);
      try
//        fFirestoreObject := Storage.UploadSynchronousFromStream(fs, ObjectName,ContentType);
//        ShowFirestoreObject;
        Storage.UploadFromStream(fs, ObjectName, ContentType, nil, nil);
      finally
//        fs.Free;
      end;
    finally
//      Storage.Free;
    end;
  end;

3. Compile and run the into demo 
4. Invoke the file upload to witness the error

**Desktop (please complete the following information):**
 - OS: [Windows 10]

**Comment by SchneiderInfosystems** [comment]
I will add the asynchronous upload asoon as possible to the intor demo project.

**Comment by gorepj** [comment]
Thank you Christoph

**Comment by SchneiderInfosystems** [comment]
I can reproduce this bug now. TFirebaseStorage.UploadFromStream need to be fixed. Thank you gorepj!

**Comment by SchneiderInfosystems** [comment]
I found the root cause of this issue. 

The bad news is the failure is in the Delphi RTL library in the REST.Client.pas.
I have entered an issue in Embarcaderos quality central:
https://quality.embarcadero.com/browse/RSP-23318

The good news is I found a workaround by changing the calling parameters for TRestRequest by changing move one part of the base URL into the Resource Parameters. I will commit and push this solution soon.
